src/ResourceManager/RecoveryServices.SiteRecovery/Commands.RecoveryServices.SiteRecovery/ReplicationProtectedItem/SetAzureRmRecoveryServicesAsrReplicationProtectedItem.cs

```diff
@@ -329,6 +341,33 @@ public override void ExecuteSiteRecoveryCmdlet()
                     }
                     vMNicInputDetailsList = getNicListToUpdate(providerSpecificDetails.VmNics);
                 }
+                else if (provider is A2AReplicationDetails)
+                {
+                    A2AReplicationDetails providerSpecificDetails = (A2AReplicationDetails)replicationProtectedItemResponse.Properties.ProviderSpecificDetails;
+
+                    if (string.IsNullOrEmpty(this.RecoveryResourceGroupId))
```

Comment:
this will as well for us.Updating as suggested

!this.MyInvocation.BoundParameters.ContainsKey(
                            Utilities.GetMemberName(() => this.RecoveryCloudServiceId))